src/OpenApi/Paths.hs

```diff
@@ -0,0 +1,80 @@
+module OpenApi.Paths where
+
+import GHC.Generics (Generic)
+import OpenApi.JSON
+import OpenApi.Prelude
+
+import qualified Data.Aeson      as JSON
+import qualified Data.Map.Strict as Map
+import qualified GHC.Enum        as GHC
+
+data Operation = Operation
+  { description :: OperationDescription
+  , operationId :: OperationId
+  , parameters  :: Maybe [Parameter]
```

Comment:
I realize this is just a code move, but is it valid to have parameters equal to `[]`? Would it be handled differently from `Nothing`?

If it is impossible for it to be `[]` perhaps you could use the `Data.List.NonEmpty` type instead?

Reply:
Yes, there are empty parameters.

Reply:
Think about "unique" (through their context) endpoints such as `GET /session`.